zone/client.cpp

```diff
@@ -10742,22 +10742,24 @@ void Client::SaveSpells()
 
 void Client::SaveDisciplines()
 {
-	std::vector<CharacterDisciplinesRepository::CharacterDisciplines> character_discs = {};
+	std::vector<CharacterDisciplinesRepository::CharacterDisciplines> v;
 
-	for (int index = 0; index < MAX_PP_DISCIPLINES; index++) {
-		if (IsValidSpell(m_pp.disciplines.values[index])) {
-			auto discipline = CharacterDisciplinesRepository::NewEntity();
-			discipline.id      = CharacterID();
-			discipline.slot_id = index;
-			discipline.disc_id = m_pp.disciplines.values[index];
-			character_discs.emplace_back(discipline);
+	for (int slot_id = 0; slot_id < MAX_PP_DISCIPLINES; slot_id++) {
+		if (IsValidSpell(m_pp.disciplines.values[slot_id])) {
+			auto e = CharacterDisciplinesRepository::NewEntity();
+
+			e.id      = CharacterID();
+			e.slot_id = slot_id;
+			e.disc_id = m_pp.disciplines.values[slot_id];
+
+			v.emplace_back(e);
 		}
 	}
 
-	CharacterDisciplinesRepository::DeleteWhere(database, fmt::format("id = {}", CharacterID()));
```

Comment:
Why is this now DeleteOne? Shouldn't we remove this line altogether if we're using ReplaceMany ?

Reply:
`DeleteOne` actually deletes everything for that character ID, similar to `DeleteWhere`, but without needing anything but the `CharacterID()` passed to it. And yeah, we can definitely remove it since we're replacing now.

Reply:
`DeleteOne` makes the assumptions that `id` is an auto incr id usually which doesn't have more than one. It's fine though I see what you're saying

Reply:
Yeah, in this case our primary key doesn't auto increment so some tables let us use `DeleteOne` as a short-hand.

Reply:
It just doesn't read well